projects/kit/components/routable-dialog/routable-dialog.component.ts

```diff
@@ -51,3 +62,19 @@ export class TuiRoutableDialogComponent {
         });
     }
 }
+
+function isFunction(x: any): boolean {
+    return (
+        // eslint-disable-next-line no-nested-ternary
+        (typeof x === 'function'
+            ? // eslint-disable-next-line no-nested-ternary
+              x.prototype
+                ? Object.getOwnPropertyDescriptor(x, 'prototype')?.writable
+                    ? 'function'
+                    : 'class'
+                : x.constructor.name === 'AsyncFunction'
+                  ? 'async'
+                  : 'arrow'
+            : '') === 'function'
```

Comment:
Looks overcomplicated(

Maybe just `isThenable` ?

https://github.com/taiga-family/maskito/blob/73b3008666ba3ccab0053d6141116a27c1e05b76/projects/react/src/lib/useMaskito.ts#L11-L13

Reply:
`() => import('etc')` it's a function, not a Promise

Reply:
fixed